### ISSUE TITLE
[kube-prometheus-stack] Chart version 47.2.0 broke compatibility (commit 990fcbc)

### ISSUE THREAD

**Issue by andrea-amadei** [prometheus-community/helm-charts#3544]
### Describe the bug a clear and concise description of what the bug is.

Newest version of kube-prometheus-stack helm chart (47.2.0) fails to install. Works as intended in version 47.1.0.

This is the error:

Error: INSTALLATION FAILED: create: failed to create: Secret "sh.helm.release.v1.prometheus.v1" is invalid: data: Too long: must have at most 1048576 bytes

### What's your helm version?

v3.12.1+gf32a527

### What's your kubectl version?

Client Version: v1.25.9 / Kustomize Version: v4.5.7 / Server Version: v1.26.4+k3s1

### Which chart?

kube-prometheus-stack

### What's the chart version?

47.2.0

### What happened?

Helm installation failed with the following error:

Error: INSTALLATION FAILED: create: failed to create: Secret "sh.helm.release.v1.prometheus.v1" is invalid: data: Too long: must have at most 1048576 bytes

### What you expected to happen?

Helm installation should have be successful. Currently working on 47.1.0.

### Enter the changed values of values.yaml?

prometheus:
  prometheusSpec:
    scrapeInterval: "15s"
    replicas: 3

    serviceMonitorSelectorNilUsesHelmValues: false

    additionalScrapeConfigs:
      <REDACTED>

alertmanager:
  alertmanagerSpec:
    replicas: 3

grafana:
  defaultDashboardsTimezone: Europe/Finland

  adminPassword: admin1

  additionalDataSources:
    - name: Loki
      type: loki
      url: http://loki-loki-distributed-query-frontend.logging:3100

### Enter the command that you execute and failing/misfunctioning.

```bash
helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
helm repo update
helm install prometheus prometheus-community/kube-prometheus-stack --version 47.2.0 -f prometheus-config.yaml --namespace monitoring --create-namespace
```

**Comment by danports** [comment]
This seems to be a known issue with Helm and one they have no immediate plans to fix - multiple issues have been reported but see for example https://github.com/helm/helm/issues/11493. I'm guessing that the kube-prometheus-stack chart has been hovering around the 1MB limit for some time now and #3083 finally pushed it over the edge.

**Comment by jkroepke** [comment]
https://github.com/prometheus-community/helm-charts/pull/3546 is a potential fix for this.

Its not the first time, that `kube-prometheus-stack` hits this limit and a fix was implemented in #3416 to lower the release size which worked well.

However, in #3543, an older version of the dashboard python script was used, and some KBs of whitespaced added again to kube-prometheus-stack which re-introduced the issue again.
